### PR TITLE
Enable new finalization by default

### PR DESCRIPTION
**Pull request by guillep**
Make ephemerons the default and only finalization mechanism.
Remove support for weak object finalization => they require full scan of elements to identify finalizers.

Many cleanups of unused variables and related methods.

### CODE REVIEW

File: smalltalksrc/VMMaker/SpurMemoryManager.class.st [hunk 1]
```diff
@@ -9626,7 +9626,11 @@ SpurMemoryManager >> popObjStack: objStack [
 						[MarkStackRecord addLast: {#EMPTY. nil}]]].
 		^nil].
 	topx := topx - 1.
+	
+	"Get the top and mark it as BADFOOD to indicate it's not available"
 	top := self fetchPointer: topx + ObjStackFixedSlots ofObject: objStack.
+	self storePointer: topx + ObjStackFixedSlots ofObject: objStack withValue: 16rBADF00D.
```
Comment (privat) [inline smalltalksrc/VMMaker/SpurMemoryManager.class.st:9666]: why marking it? Is this indented only for the simulation or debugging?

Comment (guillep) [reply smalltalksrc/VMMaker/SpurMemoryManager.class.st:9666]: Debugging ^^
I'm going to put it in the comment

File: smalltalksrc/VMMaker/SpurMemoryManager.class.st [hunk 2]
```diff
@@ -9638,7 +9634,11 @@ SpurMemoryManager >> popObjStack: objStack [
 						[MarkStackRecord addLast: {#EMPTY. nil}]]].
 		^nil].
 	topx := topx - 1.
+	
+	"Get the top and mark it as BADFOOD to indicate it's not available"
```
Comment (guillep) [inline smalltalksrc/VMMaker/SpurMemoryManager.class.st]:
```suggestion
	"Get the top and mark it as BADFOOD to indicate it's not available.
	Mark it for debugging purposes. No accesses to BADFOOD are expected."
```

**Comment by guillep** [comment]
CI is ok!

**Pull request merged**